### DISCUSSION TITLE
jenkins-docker-image-build

### DISCUSSION THREAD

**Discussion by naveenalla3000** [community/community#145441] [Programming Help] [answered]
### Body

inorder to build **docker image and push** that image to docker hub, using **Jenkins Pipeline** on instance(in my case that is docker-container of jenkins/jenkins image on my local laptop or sometimes aws-ec2 (ubuntu)), do docker need to install and run????
looking for clearfication

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by toygma** [comment, marked as answer]
To build and upload Docker images to Docker Hub, you need to have Docker installed. If Jenkins is running in a Docker container, you may need to connect a Docker socket. If you are using a server like AWS EC2, you can install Docker directly on that server.

https://medium.com/devops-technical-notes-and-manuals/how-to-run-jenkins-using-docker-on-aws-ec2-ubuntu-instance-step-by-step-guide-for-junior-devops-766d382ec86e

https://stackoverflow.com/questions/78006134/docker-container-running-as-per-logs-from-jenkins-pipeline-but-when-going-to-ec2
